client/components/deposits-overview/index.tsx

```diff
@@ -35,6 +35,9 @@ const DepositsOverview: React.FC = () => {
 		overview,
 		isLoading: isLoadingOverview,
 	} = useSelectedCurrencyOverview();
+	const isDepositsUnrestricted =
+		wcpaySettings.accountStatus.deposits?.restrictions ===
```

Comment:
Why do you check for `wcpaySettings.accountStatus.deposits.restrictions` instead of `account.deposits_blocked`?

If I understand it correctly, `account.deposits_blocked` takes account new account waiting period and `wcpaySettings.accountStatus.deposits.restrictions` is not.

Just curious.

Current code would work because [the deposits card won't show if available balance is 0 and there is no deposits](https://github.com/Automattic/woocommerce-payments/blob/251fb285089a073f6d7f7279f34b56c6352a6ae9/client/components/deposits-overview/index.tsx#L86-L89) anyway. When in waiting period, merchant won't have deposits.

Reply:
AFAICT `account.deposits_blocked` [only checks for deposits blocked](https://github.com/Automattic/woocommerce-payments-server/blob/4f187b11eb9ca3237e4806a2f7de406a6f0d53a7/server/wp-content/rest-api-plugins/endpoints/wcpay/class-deposits-controller.php#L238), and doesn't take into consideration schedule restrictions.

In this case, we want to check for both schedule restrictions and blocked deposits, which `wcpaySettings.accountStatus.deposits.restrictions === 'deposits_unrestricted'` covers.

I could include a check for `wcpaySettings.accountStatus.deposits.completed_waiting_period === true`, that might be more explicit. What do you think?

Reply:
Thank you for explaining. I don't think it's necessary to check waiting period.